cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -2183,81 +2183,196 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             let ty = ty.unwrap();
 
             let input_ty = ctx.input_ty(insn, 0);
-            match input_ty {
-                types::I8 | types::I16 | types::I32 => {
-                    // Conversion from an unsigned int smaller than 64-bit is easy: zero-extend +
-                    // do a signed conversion (which won't overflow).
-                    let opcode = if ty == types::F32 {
-                        SseOpcode::Cvtsi2ss
-                    } else {
-                        assert_eq!(ty, types::F64);
-                        SseOpcode::Cvtsi2sd
-                    };
+            if !ty.is_vector() {
+                match input_ty {
+                    types::I8 | types::I16 | types::I32 => {
+                        // Conversion from an unsigned int smaller than 64-bit is easy: zero-extend +
+                        // do a signed conversion (which won't overflow).
+                        let opcode = if ty == types::F32 {
+                            SseOpcode::Cvtsi2ss
+                        } else {
+                            assert_eq!(ty, types::F64);
+                            SseOpcode::Cvtsi2sd
```

Comment:
I know this isn't code you wrote, but in the old backend we were using `CVTTSI2S{S|D}` I believe. It would seem like, for both vector and scalar, we should be using the same class of instruction for both: either `CVT...` _or_ `CVTT...`?

Reply:
For reference, here is the old backend [expand_fcvt_from_uint](https://github.com/bytecodealliance/wasmtime/blob/c5a69cee9fbd767b43ddacf934cfc91ab8fc9dd5/cranelift/codegen/src/isa/x86/enc_tables.rs#L705) function with the scalar sequence. I would guess most of that is covered in `Inst::cvt_u64_to_float_seq` below.

Reply:
Oh, I see what I missed: `CVTT...` is being used for `FcvtToSintSat`, not `FcvtFromUint`--disregard :smile:.

Reply:
👍